src/main/java/org/apache/sysds/hops/rewrite/RewriteAlgebraicSimplificationDynamic.java

```diff
@@ -2398,7 +2399,9 @@ else if( bop.getOp() == OpOp2.MINUS && left.getDataType()==DataType.MATRIX
 			}
 			
 			//rewire parent-child operators if rewrite applied
-			if( ternop != null ) { 
+			if( ternop != null ) {
+				if (right.getForcedExecType() == Types.ExecType.FED)
+					ternop.setForcedExecType(Types.ExecType.FED);
```

Comment:
This is necessary, because our planner might assume the output to be federated. The fused ternary instruction is generated to execute in CP mode. We could either stop those rewrites from the beginning or be careful where we have to add the fed execution mode.